### PR TITLE
Catalog of my things V1.0

### PR DESCRIPTION
**Pull request by andrianarivo**
# The following features were implemented in this project:
<br/>

- [x] **Each class from the [UML class diagram](https://github.com/microverseinc/curriculum-ruby/blob/main/group-capstone/images/catalog_of_my_things.png) created in a separate .rb file.**

- [x] **All associations (1-to-many relationships) and aggregations (parent-child relationships) between classes implemented.**

- [x] **All properties visible in the diagram are defined and set up in the constructor method. Exception: properties for the 1-to-many relationships should NOT be set in the constructor method. Instead, they should have a custom setter method created.**

- [x] All methods visible in the diagram are implemented:
   - add_item method (in all classes that have that method)
     - [x] should take an instance of the Item class as an input.
     - [x]  should add the input item to the collection of items.
     - [x]  add self as a property of the item object (by using the correct setter from the item object).
 
  - can_be_archived?() in the Item class
     - [x] should return true if published_date is older than 10 years.
     - [x] otherwise, it should return false.
    
  - move_to_archive() in the Item class
     - [x] should reuse can_be_archived?() method.
     - [x]  should change the archived property to true if the result of the can_be_archived?() method is true.
     - [x] should do nothing if the result of the can_be_archived?() method is false.
 
  - can_be_archived?() in the Book class
    - [x] should override the method from the parent class.
    - [x]  should return true if parent's method returns true OR if cover_state equals to "bad".
    - [x]  otherwise, it should return false.

  - can_be_archived?() in the MusicAlbum class
     - [x] should override the method from the parent class.
     - [x]  should return true if parent's method returns true AND if on_spotify equals true.
     - [x] otherwise, it should return false.

  - can_be_archived?() in the Movie class
    - [x] should override the method from the parent class. 
    - [x] should return true if parent's method returns true OR if silent equals true.
    - [x]  otherwise, it should return false.

  - can_be_archived?() in the Game class
    - [x]  should override the method from the parent class.
    - [x]  should return true if parent's method returns true AND if last_played_at is older than 2 years.
    - [x]  otherwise, it should return false.

- [x] Added unit tests for all implemented methods.

**User interface**

- [x] Created a main.rb file that will serve as the console app entry-point.

- Our console app, at the start, is: 
  - [x] Presenting the user with a list of options to perform.
  - [x] Lets users choose an option.
  - [x] If needed, asks for parameters for the option.
  - [x] Has a way to quit the app.

- The following options are available:
   - [x] List all books
   - [x]  List all music albums
   - [x]  List of games
   - [x]  List all genres (e.g 'Comedy', 'Thriller')
   - [x]  List all labels (e.g. 'Gift', 'New')
   - [x]  List all authors (e.g. 'Stephen King')
   - [x]  Add a book
   - [x]  Add a music album
   - [x]  Add a game

 - [x] All data are preserved by saving collections in .json files.

 **Database**

- [x] Created a schema.sql file with tables that will be analogical to the structure of the classes in your app. As you cannot implement inheritance in the database tables - add all properties and associations from the parent Item class as table columns to all tables based on the child classes. Thanks to that it can be used to store data in the future.

### CODE REVIEW

File: database/schema.sql [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+  last_played_at DATE,
+  FOREIGN KEY(id) REFERENCES item(id)
+);
+
+CREATE TABLE author(
+  id INT GENERATED ALWAYS AS IDENTITY,
+  first_name VARCHAR(255),
+  last_name VARCHAR(255),
+  PRIMARY KEY(id)
+);
```
Comment (ShadyShawkat) [inline database/schema.sql:87]: - [ ] Kindly add indexes for each of the created foreign keys.

**Review by ShadyShawkat — CHANGES_REQUESTED** [review]
## Hi Team, 

Good job so far! 👍🏻 
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight
- Correct use of GitFlow ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 

Image

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: database/schema.sql (hunk 1 above)
Comment (andrianarivo) [reply database/schema.sql:87]: Thank you @ShadyShawkat for your review, we'll do the adjustments

**Review by joel-kalema — APPROVED** [review]
Hi TEAM

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# Congratulations! 🎉 


## Highlights

The project is running successfully 👍 
PR is well descriptive 👍 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me **@joel-kalema** in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**